core/capabilities/examples/on_demand_trigger.go

```diff
@@ -0,0 +1,77 @@
+package examples
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities"
+)
+
+var info = capabilities.MustCapabilityInfo(
+	"on-demand-trigger",
+	capabilities.CapabilityTypeTrigger,
+	"An example on-demand trigger.",
+	"v1.0.0",
+)
+
+// OnDemandTrigger is an example on-demand trigger.
```

Comment:
Can we explain that this trigger supports sending ad-hoc events to registered listeners?